src/Parsers/Kusto/ParserKQLProjectRename.h

```diff
@@ -19,9 +35,31 @@ class ParserKQLProjectRename : public ParserKQLBase
     const char * getName() const override { return "KQL project-rename"; }
     bool parseImpl(Pos & pos, ASTPtr & node, Expected & expected) override;
 
-private:
-    String getRenameExprFromToken(Pos & pos)
+public:
+    bool checkDuplicateAlias(const ASTPtr & node, const std::string & alias)
     {
+        (void)node;
+        (void)alias;
+
+        if (kql_context.context.has_value())
+        {
+            if (auto * select_query = node->as<ASTSelectQuery>(); !select_query->select())
+                setSelectAll(*select_query);
+            const auto sample_block = InterpreterSelectWithUnionQuery::getSampleBlock(wrapInSelectWithUnion(node), kql_context.context.value());
+            const auto & names_and_types = sample_block.getNamesAndTypes();
+            for (int i = 0; i < std::ssize(names_and_types); ++i)
+            {
+                if (names_and_types[i].name == alias)
+                    return true;
+            }
```

Comment:
You could also go with a solution based on C++20 ranges:
```suggestion
            return std::ranges::contains(names_and_types | std::views::transform([](const auto & name_and_type) { return name_and_type.name; }), alias);
```

Reply:
This doesn't seem to work in clang 15 without `-fexperimental-library`

Reply:
I'll use `std::ranges::find` in this code instead

Reply:
It seems it isn't yet implemented in Clang 16, so we can just go with `std::ranges::find` for now.